src/plugins/intel_cpu/src/nodes/executors/acl/acl_eltwise.hpp

```diff
@@ -73,28 +75,98 @@ class AclEltwiseExecutorBuilder : public EltwiseExecutorBuilder {
                 return false;
         }
 
-        // ACL supports only U8 precision on output for comparison operations
-        if (one_of(eltwiseAttrs.algorithm, Algorithm::EltwiseEqual, Algorithm::EltwiseNotEqual, Algorithm::EltwiseGreater,
-                                           Algorithm::EltwiseGreaterEqual, Algorithm::EltwiseLess, Algorithm::EltwiseLessEqual)) {
-            if (dstDescs[0]->getPrecision() != InferenceEngine::Precision::U8) {
-                return false;
+        auto checker = [](const std::vector<MemoryDescPtr>& srcDescs,
+                          const std::vector<MemoryDescPtr>& dstDescs,
+                          std::vector<Precision> srcVecPrc, Precision dstPrc) -> bool {
+            for (int i = 0; i < srcDescs.size(); i++) {
+                if (srcDescs[i]->getPrecision() != srcVecPrc[i]) return false;
             }
-        }
-        for (const auto &srcD : srcDescs) {
-            for (const auto &dstD : dstDescs) {
-                if ((srcD->getPrecision() != InferenceEngine::Precision::FP32 &&
-                     srcD->getPrecision() != InferenceEngine::Precision::FP16) ||
-                     srcD->getPrecision() != dstD->getPrecision())
+            if (dstDescs[0]->getPrecision() != dstPrc) { return false; }
+            return true;
+        };
+
+        switch (eltwiseAttrs.algorithm) {
```

Comment:
Can you unite both switch cases into one? I belive the list of cases should be the same. Just call `return false` inside `default` case here.